runtime/src/platform_utils.cpp

```diff
@@ -0,0 +1,40 @@
+#include "platform_utils.hpp"
+
+#if __linux__
+#include <sched.h>
+
+int my_sched_getcpu() {
+    return ::sched_getcpu();
+}
+
+#elif __APPLE__
+#include <sys/sysctl.h>
+#include <pthread.h>
+
+int my_sched_getcpu() {
+    int cpu = -1;
+    size_t len = sizeof(cpu);
+
+    if (sysctlbyname("hw.cpulocation", &cpu, &len, NULL, 0) == -1) {
+        return -1;  // Handle error
```

Comment:
Call SPDLOG_ERROR to log errors

Reply:
okay, will do this change

Reply:
have added a `SPDLOG_ERROR` and am also returning -1 to be handled in other functions wherever it is called.